src/main/scala/io/iohk/ethereum/jsonrpc/TestJsonMethodsImplicits.scala

```diff
@@ -0,0 +1,110 @@
+package io.iohk.ethereum.jsonrpc
+
+import akka.util.ByteString
+import io.iohk.ethereum.jsonrpc.JsonRpcController.{JsonDecoder, JsonEncoder}
+import io.iohk.ethereum.jsonrpc.JsonRpcErrors.InvalidParams
+import io.iohk.ethereum.jsonrpc.TestService._
+import org.json4s.JsonAST._
+import org.json4s.JsonDSL._
+import org.spongycastle.util.encoders.Hex
+
+import scala.util.Try
+
+object TestJsonMethodsImplicits extends JsonMethodsImplicits {
+
+  implicit val test_setChainParams = new JsonDecoder[SetChainParamsRequest] with JsonEncoder[SetChainParamsResponse] {
+    private def extractAccounts(accountsJson: JValue): Either[JsonRpcError, Map[ByteString, AccountConfig]] = {
+      val accounts = accountsJson.asInstanceOf[JObject].values.collect { case (key, accObj: Map[_, _]) =>
+        val rawWei = accObj.asInstanceOf[Map[String, Any]]("wei").asInstanceOf[String]
+        val wei = if (rawWei.startsWith("0x")) BigInt(rawWei.replace("0x", ""), 16)
+        else BigInt(rawWei, 10)
+        ByteString(Hex.decode(key.replace("0x", ""))) -> AccountConfig(None, wei)
+      }
+      Right(accounts)
+    }
+
+    private def extractBlockchainParams(blockchainParamsJson: JValue): Either[JsonRpcError, BlockchainParams] = {
+      for {
+        eIP150ForkBlock <- extractQuantity(blockchainParamsJson \ "EIP150ForkBlock")
+        eIP158ForkBlock <- extractQuantity(blockchainParamsJson \ "EIP150ForkBlock")
+        accountStartNonce <- extractQuantity(blockchainParamsJson \ "EIP150ForkBlock")
```

Comment:
this code looks a little suspicious to me. Is this a copy/paste error or was it deliberate?

Reply:
lol! I totally overlooked that.